[docs] remove "development mode" qualifier from navigation devtools plugin (#27102)
expo/expo: docs/pages/debugging/devtools-plugins.mdx

Expo provides some dev tools plugins for common debugging tasks. Follow the instructions below to start using them in your app.

> > Each of these dev tools plugin hooks will only enable the plugin in development mode, so you don't need to worry about this code affecting your production bundle.

Reviewer: ```suggestion
> **Note**: Each of the following dev tools plugin hooks will only enable the plugin in development mode. It doesn't affect your production bundle.
```
Let's make it a bit more concise and the callout header such as "Note".
